src/sys.d.ts

```diff
@@ -0,0 +1,1352 @@
+export type JsObj = Obj | number | string | boolean | ((...args: any) => any)
+
+export class Obj {
+  equals(that: JsObj | null): boolean
+  compare(that: JsObj): number
+  hash(): number
+  toStr(): string
+  trap(name$: string, args?: List<JsObj | null> | null): JsObj | null
```

Comment:
Here `args` is already nullable and optional, there is no need for `| null` here.